### ISSUE TITLE
iCloud sync doesn't always work

### ISSUE THREAD

**Issue by LordFransie** [Aidoku/Aidoku#111]
When I have two devices and I set iCloud backup to both, the iCloud backups do not keep the two instances in sync

**Comment by Skittyblock** [comment]
iCloud sync has a number of issues that still need to be ironed out. It appears to work for some people and not others with no apparent cause or reason. I know at least that iCloud won't work when sideloading, so if you're not running the TestFlight build on all your devices then that might be your issue.

**Comment by Jamerrone** [comment]
Just wanted to share my experience in case it’s relevant. Sync seems to work for me, however, it does not add titles to my library even though I can see the ribbon icon. So the app thinks it added to the library, but it did not, this only happens on secondary devices (e.g. the device where I did not manually add to my library, meaning the device I want to sync).

**Comment by Skittyblock** [comment]
hmm... so a title won't show in your library tab but if you look at it in the source it'll show the bookmark icon on the cover? is the bookmark button highlighted in the manga page?

**Comment by Jamerrone** [comment]
Yes, exactly as you described, and yes the button is highlighted and the ribbon is visible but the library tab is empty. Maybe important to note that the main device is an iPhone while de secondary device is an MacBook Pro.

**Comment by Jamerrone** [comment]
Actually, please ignore my comment. I just wanted to record my screen to show the bug, and of course it worked just fine… Lovely, I have had this issue for 2 days now, restarted both machines, reinstalled the apps, it never worked. Now that I wanted to share a recording, everything magically worked. Welp, I am glad, but I have no idea what changed.

**Comment by LordFransie** [comment]
I've marked this close as for all intent and purposes I can no longer replicate this issue.

**Comment by Skittyblock** [comment]
reopening this because I've still seen others with issues related to iCloud sync

**Comment by Radicitus** [comment]
I'm also having issues with this and using iCloud sync was a big reason I reserved a spot on the TestFlight. Any idea when this could be fixed? I'm also willing to look into a fix although I'm new to mobile dev

**Comment by AshenPaladin** [comment]
It worked well for me for almost two years, and then it suddenly stopped working. It worked partially for a while - syncing either the history, or the library, but not the two together, and now it seems to have stopped working altogether. Even if the app is seen by iCloud in settings and it reports having data in iCloud storage that is increasing bit by bit as it should.

**Comment by Trannosaur** [comment]
Heyo, running locally I found I had to add this line: 
localDescription.setOption(true as NSNumber, forKey: NSPersistentHistoryTrackingKey) @~CoreDataManager.swift:60

To be honest, I haven't read much of the cloudkit documentation, so I'm not sure that's the correct solution, it was just a random poking of the bear because I was seeing this error in the logs:
"History Change Request failed as no history tracking option detected"

**Comment by alsheikhhusam** [comment]
+1 Cloud Sync does not work for me

**Comment by NeKoOuO** [comment]
iCloud doesn't work for me too, and also make me cannot open the app

**Comment by Skittyblock** [comment]
it only works when installed through testflight

**Comment by alsheikhhusam** [comment]
@Skittyblock I am using testflight and still doesn't work

**Comment by Weishao666** [comment]
iCloud doesn't work for me too

**Comment by AshenPaladin** [comment]
It was working for me on Testflight without any issues for the better part of a year, but with the update to 0.7 it's again stopped working properly. The history seems to sync a little, but the library doesn't sync at all and in any case the history sync stops working after that initial bit of syncing.

It was working without any issues as early as yesterday so it's definitively the update to 0.7 that broke it, and a few other people have reported this also on the Discord.

**Comment by dyphire** [comment]
Similarly, TF's iCloud synchronization doesn't work for me.
